CMakeLists.txt

```diff
@@ -1,26 +1,157 @@
-cmake_minimum_required(VERSION 3.0)
+#TODO: write FindxxHash for libxxhash-dev compatibility?
+#Cmake for xxHash is unofficial, so libxxhash-dev doesn't have xxHashConfig.cmake
+
+cmake_minimum_required(VERSION 3.1)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS ON)
+
 project(test)
 
-find_package(xxHash 0.7 CONFIG REQUIRED)
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+  message("Adding GNU compiler flags")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -W -Wall -O3")
```

Comment:
what do these compiler flags do? do we want to just display warnings or treat warnings as errors as well?

Reply:
They're the exact same compiler flags that we're currently using.
`-Wall` turns on most warnings
`-W` is an older name for -Wextra, turns on extra warnings
`-O3` is to optimize code a lot

In our current CMakeLists we also have `-std=c++11`, but I changed that to setting [`CXX_STANDARD`](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html)

If we want to make all warnings as errors we could add -Werror, but as of right now I get a bunch of warnings when compiling our code so that's probably not a good idea

Reply:
If we're going to turn on warnings but ignore them, why have them on at all?

Reply:
I'm confused what you're saying here. Are you saying we should fix the warnings or not?

Reply:
Can you add a task for that?